In [2]:
from ultralytics import YOLO
import time

In [3]:
start = time.time()
model = YOLO("yolov8n.yaml")
model = YOLO("yolov8n.pt")
end = time.time()-start


                   from  n    params  module                                       arguments                     
  0                  -1  1       464  ultralytics.nn.modules.Conv                  [3, 16, 3, 2]                 
  1                  -1  1      4672  ultralytics.nn.modules.Conv                  [16, 32, 3, 2]                
  2                  -1  1      7360  ultralytics.nn.modules.C2f                   [32, 32, 1, True]             
  3                  -1  1     18560  ultralytics.nn.modules.Conv                  [32, 64, 3, 2]                
  4                  -1  2     49664  ultralytics.nn.modules.C2f                   [64, 64, 2, True]             
  5                  -1  1     73984  ultralytics.nn.modules.Conv                  [64, 128, 3, 2]               
  6                  -1  2    197632  ultralytics.nn.modules.C2f                   [128, 128, 2, True]           
  7                  -1  1    295424  ultralytics.nn.modules.Conv                  [128

In [4]:
f'time: {end}'

'time: 0.30501627922058105'

In [6]:
start = time.time()
model.train(data="coco128.yaml", epochs=5)
end = divmod(time.time()-start, 60)

Ultralytics YOLOv8.0.56  Python-3.9.2 torch-2.0.0+cu117 CUDA:0 (NVIDIA GeForce RTX 3060, 12287MiB)
yolo\engine\trainer: task=detect, mode=train, model=yolov8n.pt, data=coco128.yaml, epochs=5, patience=50, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=False, optimizer=SGD, verbose=True, seed=0, deterministic=True, single_cls=False, image_weights=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, hide_labels=False, hide_conf=False, vid_stride=1, line_thickness=3, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=F

                person        128        254      0.847      0.709      0.802      0.585
               bicycle        128          6       0.95      0.333      0.464       0.39
                   car        128         46      0.685      0.261      0.407      0.226
            motorcycle        128          5      0.829        0.8       0.92       0.71
              airplane        128          6      0.829          1      0.948      0.832
                   bus        128          7       0.78      0.714      0.722      0.651
                 train        128          3      0.749          1      0.995      0.852
                 truck        128         12      0.798        0.5      0.588      0.395
                  boat        128          6      0.523      0.372      0.557      0.329
         traffic light        128         14      0.823      0.214      0.223      0.143
             stop sign        128          2      0.802          1      0.995      0.647
                 benc

In [7]:
f'train time: {end}'

'train time: (6.0, 1.5228426456451416)'

In [ ]:
start2 = time.time()
metrics = model.val()
end2 = divmod(time.time()-start, 60)

Ultralytics YOLOv8.0.56  Python-3.9.2 torch-2.0.0+cu117 CUDA:0 (NVIDIA GeForce RTX 3060, 12287MiB)
YOLOv8n summary (fused): 168 layers, 3151904 parameters, 0 gradients, 8.7 GFLOPs

Dataset 'coco.yaml' images not found , missing paths ['C:\\Users\\rhayr\\Downloads\\Compressed\\datasets\\coco\\val2017.txt']
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 169M/169M [00:08<00:00, 21.9MB/s]
Unzipping C:\Users\rhayr\Downloads\Compressed\datasets\coco2017labels-segments.zip to C:\Users\rhayr\Downloads\Compressed\datasets...
Unzipping C:\Users\rhayr\Downloads\Compressed\datasets\coco\images\val2017.zip to C:\Users\rhayr\Downloads\Compressed\datasets\coco\images...


In [4]:
start = time.time()
results = model.predict('C:\\Users\\rhayr\\Downloads\\Compressed\\ultralytics\\Images\\test')
end = divmod(time.time()-start, 60)
print(end)


image 1/10 C:\Users\rhayr\Downloads\Compressed\ultralytics\Images\test\000000000139.jpg: 448x640 1 person, 5 chairs, 1 potted plant, 2 dining tables, 1 tv, 1 refrigerator, 1 clock, 1 vase, 96.0ms
image 2/10 C:\Users\rhayr\Downloads\Compressed\ultralytics\Images\test\000000000285.jpg: 640x608 1 bear, 96.0ms
image 3/10 C:\Users\rhayr\Downloads\Compressed\ultralytics\Images\test\000000000632.jpg: 512x640 1 bottle, 1 chair, 2 potted plants, 1 bed, 2 books, 91.0ms
image 4/10 C:\Users\rhayr\Downloads\Compressed\ultralytics\Images\test\000000000724.jpg: 640x480 1 truck, 2 stop signs, 98.0ms
image 5/10 C:\Users\rhayr\Downloads\Compressed\ultralytics\Images\test\000000000776.jpg: 640x448 4 teddy bears, 94.0ms
image 6/10 C:\Users\rhayr\Downloads\Compressed\ultralytics\Images\test\000000000785.jpg: 448x640 1 person, 1 skis, 7.0ms
image 7/10 C:\Users\rhayr\Downloads\Compressed\ultralytics\Images\test\000000000802.jpg: 640x448 1 oven, 1 refrigerator, 8.0ms
image 8/10 C:\Users\rhayr\Downloads\Compr

(0.0, 6.878409147262573)


Speed: 0.2ms preprocess, 58.9ms inference, 46.2ms postprocess per image at shape (1, 3, 640, 640)
Speed: 0.5ms preprocess, 60.9ms inference, 46.8ms postprocess per image at shape (1, 3, 640, 640)
Speed: 0.4ms preprocess, 60.8ms inference, 46.9ms postprocess per image at shape (1, 3, 640, 640)